integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -1023,31 +1023,39 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
     dataType match {
       case "string" =>
         Field(field.column, Some("String"), field.name, Some(null), field.parent,
-        field.storeType, field.schemaOrdinal, field.precision, field.scale, field.rawSchema
+        field.storeType, field.schemaOrdinal, field.precision, field.scale, field.rawSchema,
```

Comment:
Instead of changing Field can you add another class for extra information and wrap the Field into it. Better try t avoid old classes. This Field already has lot of paramaters